pkg/client/typed/discovery/discovery_client.go

```diff
@@ -67,6 +67,10 @@ type ServerResourcesInterface interface {
 	ServerResourcesForGroupVersion(groupVersion string) (*unversioned.APIResourceList, error)
 	// ServerResources returns the supported resources for all groups and versions.
 	ServerResources() (map[string]*unversioned.APIResourceList, error)
+	// ServerAllResources returns the supported resources for all groups and versions.
+	ServerAllResources() ([]unversioned.GroupVersionResource, error)
+	// ServerAllNamespacedResources returns the supported namespaced resources for all groups and versions.
+	ServerAllNamespacedResources() ([]unversioned.GroupVersionResource, error)
 	// ServerPreferredResources returns the supported resources with the version preferred by the
 	// server.
 	ServerPreferredResources() ([]unversioned.GroupVersionResource, error)
```

Comment:
Yeah, I think with the current implementation it makes sense to extend previous method, @krousey any disagreement on that?